cmd/orch-tester/orch_tester.go

```diff
@@ -549,3 +599,209 @@ func (s *streamerClient) getOrchestrators() ([]*orch, error) {
 
 	return data.Data.Transcoders, nil
 }
+
+// Get a random source/rendition sample segment and write it to external storage
+func (s *streamerClient) randomSample(ctx context.Context, mid string, orch string) error {
+	randSeqNo := rand.Int() % numSegments
+	randProfile := s.profiles[rand.Int()%len(s.profiles)]
+	//stream/manifestID/profile/SeqNo.ts
+	sourceUrl := fmt.Sprintf("%v/stream/%v_0_0/source/%v.ts", s.broadcaster, mid, randSeqNo)
+	renditionUrl := fmt.Sprintf("%v/stream/%v_0_0/%v/%v.ts", s.broadcaster, mid, randProfile, randSeqNo)
+	var (
+		rendition []byte
+		source    []byte
+		err       error
+	)
+	tick := time.NewTicker(2 * time.Second)
+	for {
+		select {
+		case <-ctx.Done():
+			glog.Info("random sampler timed out")
+			// If we haven't found the segment try to sample from whatever playlist we can
+			rendition, source, renditionUrl, err = s.sampleFromPlaylist(mid)
+			if err != nil {
+				glog.Error(err)
+			}
+			break
+		case <-tick.C:
+			glog.V(model.DEBUG).Infof("Trying to download %v", renditionUrl)
+			rendition, err = s.downloadSegment(renditionUrl)
+			if err != nil {
+				glog.V(model.DEBUG).Info(err)
+				continue
+			}
+			source, err = s.downloadSegment(sourceUrl)
+			if err != nil {
+				glog.V(model.DEBUG).Info(err)
```

Comment:
Should be Error instead of Info

Reply:
We currently retry until the stream finishes which would log the error on each retry , that might be a little bit verbose as it's not a critical error (we have a fallback)

Reply:
Ah and:

- `downloadSegment()` is supposed to return an error when the segment is not available yet
- `Verbose` type returned by `glog.V` only exposes Info based logging

I see - ok to do this for now then.

I do think that this is a bit of a code smell since an error is not included in the error log. If someone were to filter through ERROR logs, these errors would not show up. And if it's not a big deal for these errors not to show up that may be a prompt to consider whether that error is really an error (or if there is room to distinguish an expected scenario from a true error condition). Anyways, I understand why we're doing this right now, but thought this would be worth mentioning.